exporters/zipkin/src/test/java/io/opentelemetry/exporter/zipkin/ZipkinSpanExporterEndToEndHttpTest.java

```diff
@@ -170,4 +177,17 @@ private static Span buildZipkinSpan(InetAddress localAddress) {
         .putTag(ZipkinSpanExporter.OTEL_STATUS_CODE, "OK")
         .build();
   }
+
+  private static List<Span> getTrace(String traceId) {
+    AggregatedHttpResponse response =
+        client.get(zipkinUrl("/api/v2/trace/" + traceId)).aggregate().join();
+    if (response.status().equals(HttpStatus.NOT_FOUND)) {
```

Comment:
Using *just* the HTTP status for this seems dangerous as you would also get 404 if you had a typo like `api/v2/trac/`.

More importantly, shouldn't an exception be thrown here for other non-2xx responses?

Reply:
I think this is reasonable enough for the method since it's also used from all the other tests, which would fail for such a big issue.

Could add an exception for other error cases but I think it's unexpected enough to keep things simple (instead of explicit exception it'll be json parsing exception).